### PR TITLE
chore: build the dist files using webpack

### LINKED ISSUE #344: localForage 1.2.2 + webpack 1.5.3 doesn't want to collaborate
Hello. It seems that this PR don't fix problem of using localForage with webpack. I've created gist to illustrate problem https://gist.github.com/ALF-er/83325c24019de4edc851 (just place file index.html in ./src/ dir and main.js in ./src/js/ dir).
If I run "npm run dev" (it just copy index.html in ./dist/ dir, compile main.js with webpack and then run webpack-dev-server for ./dist/) I get:
```
WARNING in ./~/localforage/src/localforage.js
Critical dependencies:
332:20-336:22 the request of a dependency is an expression
 @ ./~/localforage/src/localforage.js 332:20-336:22

ERROR in ./~/localforage/src/drivers/localstorage.js
Module not found: Error: Cannot resolve module 'localforageSerializer' in C:\dev\projects\wp_lf_test\node_modules\localforage\src\drivers
 @ ./~/localforage/src/drivers/localstorage.js 70:16-59

ERROR in ./~/localforage/src/drivers/websql.js
Module not found: Error: Cannot resolve module 'localforageSerializer' in C:\dev\projects\wp_lf_test\node_modules\localforage\src\drivers
 @ ./~/localforage/src/drivers/websql.js 63:16-59
```
If in main.js uncomment second line (and comment first ofc) then it builds w/o errors but in browser console I get `localForage.getItem` - undefined function...

Maybe you can suggest something?

### LINKED ISSUE #424: Use a new generation module bundler to create the dist files
This is just to track a discussion about using a module bundler to create the dist files, instead of `grunt-contrib-concat` that we currently use.

Some points of consideration are:
* Do we have a preference on any specific bundler.
* UMD standalone package generation
  * should it be able to automatically create one
  * or we can write our own export code
* How does requiring the dist bundle works from the same/other bundlers
  * [Webpack](https://github.com/webpack/webpack) renames all require calls by default
  * [Browserify](https://github.com/substack/node-browserify) needs [derequire](https://github.com/calvinmetcalf/derequire) in order to require existing bundles
* Is it possible to require parts from within the dist bundle
  * for example can we require the serializer

### PR DESCRIPTION
**Pull request by thgreasi**
Properly Resolves #344, #412, #413.
Resolves #424.

### CODE REVIEW

**Comment by thgreasi** [comment]
Note: before releasing this, we should also add the `getDriver()` method and expose the `serializer` so that (existing or not) custom drivers have a way to use it.

File: Gruntfile.js [hunk 1]
```diff
@@ -61,16 +61,13 @@ module.exports = exports = function(grunt) {
...
-                        'build/es5src/utils/**/*.js',
-                        'build/es5src/drivers/**/*.js',
-                        'build/es5src/localforage.js'
+                        'dist/localforage.nopromises.js'
                     ],
                     'dist/localforage.nopromises.js': [
-                        'build/es5src/utils/**/*.js',
-                        'build/es5src/drivers/**/*.js',
-                        'build/es5src/localforage.js'
+                        // just to add the BANNER
```
Comment (tofumatt) [inline Gruntfile.js]: Hah, clever :smile:

**Comment by tofumatt** [comment]
Should we just expose `serializer` as a Promise that returns the serializer like `getDriver()` will return a Promise that resolves to the driver name? We could allow people to overwrite the serializer with an API or just expose it as a variable (`localforage.serializer = function() { /// custom serializer here }`).

Though, do people really ever want to overwrite the serializer? I dunno.

**Comment by thgreasi** [comment]
+1 on using a Promise based API to retrieve the serializer.
Do you prefer exposing as a Promise as a property `localforage.serializer` or a method called `getSerializer` and be more consistent with `getDriver()`? I would expect that a property would work like a namespace and just be there ready & waiting for me to use, instead of having to using `.then()`.

I guess that `getDriver()` and exposing the serializer should be a separate PR from this. I have already implemented `getDriver()` in my [trySetDriver branch](https://github.com/thgreasi/localForage/tree/trysetdriver). I can open a new PR with `getDriver()` & exposing serializer right after this gets merged.

**Comment by thgreasi** [comment]
Just did a rebase on this branch over the current master.

**Comment by thgreasi** [comment]
Any feedback on this? I have some branches ready (getDriver & a workaround for #363) and waiting for this to be merged.

**Comment by tofumatt** [comment]
Oops, right. I'll check it out after lunch today.




PS I'm on holiday (with actual time off! 🍹) this week so expect responses to be slower than usual, but I'm just in Lisbon so not like I'll never be checking my email 😉



- tofumatt

**Comment by tofumatt** [comment]
(Sorry, I got sidetracked and I'm about to fly from AMS to LIS, but I'll review this tonight or tomorrow morning!)



- tofumatt

**Comment by thgreasi** [comment]
Np hf :smile:
Vacations are always open-source productive periods anyway.

**Comment by af7** [comment]
Hi! 

For now we have to do the following for the serializer (for a custom driver):

```
        var serializerPromise = new Promise(function (resolve/*, reject*/) {
            resolve(require('./../../../../node_modules/localforage/src/utils/serializer'));
        });
```

It would be great if @thgreasi's proposal for a `getSerializer` (or a property) is implemented; that ugly code can be removed this way.

**Comment by tofumatt** [comment]
@af7 We will definitely implement a `getSerializer` method, though it will be in a different PR :smile:

**Comment by tofumatt** [comment]
Sorry, I had a real vacation and couldn't get to this until now :scream:

**Pull request merged**